### PR TITLE
ДЗ 48. Node.js - Work with filesadded file upload

### PR DESCRIPTION
**Pull request by MartaSmuk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
-package-lock.json
\ No newline at end of file
+package-lock.json
```
Comment (VitalyaIlin) [inline .gitignore]: remove `package-lock.json` from the gitignore

File: app.js [hunk 2]
```diff
@@ -1,77 +1,267 @@
...
 import { users } from './storage.js';
 import { orders } from './storage.js';
+import { ApiError } from './errorHandler.js';
+import { fileURLToPath } from 'url';
 
 const app = express();
 const PORT = 3000;
 
+//instantiate event Emitter
+class FileEventEmitter extends EventEmitter {}
```
Comment (VitalyaIlin) [inline app.js]: is it worth creating a new class?

File: app.js [hunk 3]
```diff
@@ -1,77 +1,267 @@
...
+        return res.status(400).json({ error: 'there is no file to upload'});
+    };
+})
+
+
+// register/create user
+app.post('/api/register', async (req, res) => {
+
+    try {
+        const value = await userSchema.validateAsync(req.body);
```
Comment (VitalyaIlin) [inline app.js:178]: Can you send some error message if the data is invalid?

**Review by VitalyaIlin — COMMENTED** [review]
check comments please

File: .gitignore (hunk 1 above)
Comment (MartaSmuk) [reply .gitignore]: thank you, deleted

File: app.js (hunk 2 above)
Comment (MartaSmuk) [reply app.js]: thank you, deleted class extention

File: app.js (hunk 3 above)
Comment (MartaSmuk) [reply app.js:178]: thank you, implemented error checks and error messages

**Pull request merged**